File: w11/d02/Joe_Park/penguin_pals2/app/controllers/penguins_controller.rb
class PenguinsController < ApplicationController

  respond_to :json

  def index
    penguins = Penguin.all
    respond_with penguins
  end

  def create
    penguin = Penguin.create(penguin_params)
    respond_with penguin
  end

  def destroy
    penguin = Penguin.find(params[:id])
    penguin.destroy
    respond_with penguin
  end

  def update
    penguin = Penguin.find(params[:id])
    penguin.update(penguin_params)
    respond_with penguin
  end


  private

  def penguin_params
    params.require(:penguin).permit(:name)
  end

end

File: w13/d01/INSTRUCTORS/penguin_pals_node/penguinsController.js
var Penguin = require(__dirname + '/penguin.js');

module.exports.index = function(req, res){
	Penguin.find(function(err, penguins){
		if (err) return console.error(err);
		res.send(penguins);
	})
}

module.exports.create = function(req, res){
	Penguin.create({ name: req.body.name }, function(err, penguin){
		if (err) return console.error(err);
		res.send(penguin);
	})
}

module.exports.update = function(req, res){
	Penguin.findById(req.params.id, function(err, penguin){
		if (err) return console.error(err);
		penguin.update({ name: req.body.name }, function(err, updated){
			if (err) return console.error(err);
			res.send(updated);
		})	
	})
}

module.exports.delete = function(req, res){
	Penguin.findById(req.params.id, function(err, penguin){
		if (err) return console.error(err);
		penguin.remove(function(err){
			if (err) return console.error(err);
			return res.send({ 'succes' : 'ok' });
		})
	})
}














